client/allocator/src/freeing_bump.rs

```diff
@@ -493,6 +505,17 @@ impl FreeingBumpHeapAllocator {
 		*bumper += size;
 		Ok(res)
 	}
+
+	fn observe_memory_size<M: Memory + ?Sized>(
```

Comment:
Why isn't that a method of the struct?

Reply:
I see, because of `PoisonBomb`.

However, I don't see any reason why we don't do this check before arming the bomb? Nothing will be poisoned when we return an error here.

Reply:
yeah, just out of cautiousness. The idea I followed is the arming the bomb comes always first. Can change it though

Reply:
Yeah fine, keep it that way.